### PR TITLE
Add support for --insecure flag on compliance login for profile retrieval

### PR DESCRIPTION
**Pull request by gsloan-chef**
If an Automate server is using self signed SSL cert, inspec compliance login fails with SSL error during build effortless audit package. Does not seem to support standard SSL cert file override locations during build time.

This change adds an optional "scaffold_compliance_insecure" environment variable that, when set to non-false value, will add the "--insecure" option to the inspec compliance login command during build phase.

The primary change is in scaffold.ps1 and scaffold.sh for Inspec. Also added examples and tests with the scaffold_compliance_insecure variable.

### CODE REVIEW

File: examples/effortless_audit/windows/linux/controls/example.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+title 'CIS Automate Test'
```
Comment (TheLunaticScripter) [inline examples/effortless_audit/windows/linux/controls/example.rb]: I don't think this is the right folder structure for this.

File: scaffolding-chef-inspec/lib/scaffolding.ps1 [hunk 2]
```diff
@@ -97,9 +105,11 @@ if(!`$env:CFG_CHEF_LICENSE){
 }
 `$env:CHEF_LICENSE = `$env:CFG_CHEF_LICENSE
 `$WAIVER="{{pkg.svc_config_path}}/waiver.yml"
+`$INPUTS="{{pkg.svc_config_path}}/inputs.yml"
```
Comment (TheLunaticScripter) [inline scaffolding-chef-inspec/lib/scaffolding.ps1]: This change should be in a separate PR.

File: scaffolding-chef-inspec/lib/scaffolding.ps1 [hunk 3]
```diff
@@ -116,7 +126,7 @@ function Invoke-Inspec {
     # TODO: This is set to --json-config due to the
     #  version of InSpec being used please update when InSpec is updated
     if(`$version -gt [Version]"4.17.27"){
-        {{pkgPathFor "$scaffold_inspec_client"}}/bin/inspec.bat exec `$PROFILE_PATH --config `$CONFIG --waiver-file `$WAIVER --log-level `$env:CFG_LOG_LEVEL
+        {{pkgPathFor "$scaffold_inspec_client"}}/bin/inspec.bat exec `$PROFILE_PATH --config `$CONFIG --waiver-file `$WAIVER --input-file `$INPUTS--log-level `$env:CFG_LOG_LEVEL
```
Comment (TheLunaticScripter) [inline scaffolding-chef-inspec/lib/scaffolding.ps1]: This is something different than just the insecure and should be in a different PR.

File: scaffolding-chef-inspec/lib/scaffolding.sh [hunk 4]
```diff
@@ -60,9 +60,17 @@ do_default_before() {
       build_line "$message"
       exit 1
     else
-      inspec compliance login $scaffold_automate_server_url \
+      if [ ! $scaffold_compliance_insecure ]; then
```
Comment (TheLunaticScripter) [inline scaffolding-chef-inspec/lib/scaffolding.sh]: I think this looks good is. Do you also need the insecure for the archive command or is that working cause the compliance login was insecure?

File: scaffolding-chef-inspec/lib/scaffolding.sh [hunk 5]
```diff
@@ -119,6 +127,7 @@ CFG_CHEF_LICENSE={{cfg.chef_license.acceptance}}
 CFG_CHEF_LICENSE="\${CFG_CHEF_LICENSE:-undefined}"
 CONFIG="{{pkg.svc_config_path}}/inspec_exec_config.json"
 WAIVER="{{pkg.svc_config_path}}/waiver.yml"
+INPUTS="{{pkg.svc_config_path}}/inputs.yml"
```
Comment (TheLunaticScripter) [inline scaffolding-chef-inspec/lib/scaffolding.sh]: Inputs needs to be in a separate PR>

File: scaffolding-chef-inspec/tests/linux/user-a2insecure-profile-dep/controls/example.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+title 'CIS Automate Test'
```
Comment (TheLunaticScripter) [inline scaffolding-chef-inspec/tests/linux/user-a2insecure-profile-dep/controls/example.rb]: Let's talk more about this but these are not the right test files that need to be updated. The test that need to be added for this are the ones that run through expeditor.

File: examples/effortless_audit/windows/Readme.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
+# Effortless Audit
```
Comment (TheLunaticScripter) [inline examples/effortless_audit/windows/Readme.md]: I don't think we need a duplicate readme for windows since this is similar to the readme that is already there I think it needs to be moved to the top level of the Effortless Audit examples folder.

**Comment by gsloan-chef** [comment]
Closing due to "polluted" checkin, need to resubmit with just the relevant changes. Stay tuned
